criu/pages-compress.c

```diff
@@ -89,33 +102,28 @@ static int decompress_image(int comp_fd) {
 	if (0 > decomp_fd) {
 		pr_err("Can't create file, errno=%d\n", errno);
 		LZ4F_freeDecompressionContext(dctx);
+		munmap(mapped_addr, file_stat.st_size);
```

Comment:
Not really a requirement but these cleanups in every error path can be mistakenly forgotten during updates in the future. Maybe you could provide a function(s) for the inner part and use one `munmap()` (in other case one `LZ4F_freeDecompressionContext`) in the case the inner function returns with a failure result? Not sure if it is well understandable.

Reply:
I thought something similar when implementing this. A good chance to implement it now :)